source/extensions/io_socket/user_space/file_event_impl.cc

```diff
@@ -0,0 +1,78 @@
+#include "extensions/io_socket/user_space/file_event_impl.h"
+
+#include <cstdint>
+
+#include "common/common/assert.h"
+
+#include "extensions/io_socket/user_space/io_handle.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace IoSocket {
+namespace UserSpace {
+
+FileEventImpl::FileEventImpl(Event::Dispatcher& dispatcher, Event::FileReadyCb cb, uint32_t events,
+                             IoHandle& io_source)
+    : schedulable_(dispatcher.createSchedulableCallback([this, cb]() {
+        auto ephemeral_events = event_listener_.getAndClearEphemeralEvents();
+        ENVOY_LOG(trace, "User space event {} invokes callbacks on events = {}",
+                  static_cast<void*>(this), ephemeral_events);
+        cb(ephemeral_events);
+      })),
+      io_source_(io_source) {
+  setEnabled(events);
+}
+
+void FileEventImpl::activate(uint32_t events) {
+  // Only supported event types are set.
+  ASSERT((events & (Event::FileReadyType::Read | Event::FileReadyType::Write |
+                    Event::FileReadyType::Closed)) == events);
+  event_listener_.onEventActivated(events);
+  schedulable_->scheduleCallbackNextIteration();
+}
+
+void FileEventImpl::setEnabled(uint32_t events) {
+  // Only supported event types are set.
+  ASSERT((events & (Event::FileReadyType::Read | Event::FileReadyType::Write |
+                    Event::FileReadyType::Closed)) == events);
+  event_listener_.clearEphemeralEvents();
```

Comment:
Why clear the events here? Is it the same reason as in `FileEventImpl::setEnabled()`? Please add a comment on why this is correct.

Reply:
Yes, the event owner is likely to see consistent behavior among various event implementation. Updating the comment